src/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableModelControllerBase.php

```diff
@@ -118,22 +166,7 @@ public function setAction()
             // load model and update with provided data
             $mdl = $this->getModel();
             $mdl->setNodes($this->request->getPost(static::$internalModelName));
-
-            // perform validation
-            $valMsgs = $mdl->performValidation();
-            foreach ($valMsgs as $field => $msg) {
-                if (!array_key_exists("validations", $result)) {
-                    $result["validations"] = array();
-                }
-                $result["validations"][static::$internalModelName.".".$msg->getField()] = $msg->getMessage();
-            }
-
-            // serialize model to config and save
-            if ($valMsgs->count() == 0) {
-                $mdl->serializeToConfig();
-                Config::getInstance()->save();
-                $result["result"] = "saved";
-            }
+            $result = $this->save($mdl, null, $this->setActionHook);
```

Comment:
@pv2b phalcon won't like this passing, "Notice: Access to undefined property setActionHook". We probably better split the save into validate > save if you want to hook functions.